api/src/components/user/controller.ts

```diff
@@ -32,20 +32,23 @@ export const get = async (event: I.OptionalAuthenticatedAPIRequest<undefined, un
 };
 
 export const getPublications = async (
-    event: I.OptionalAuthenticatedAPIRequest<undefined, undefined, I.GetUserParameters>
+    event: I.OptionalAuthenticatedAPIRequest<undefined, I.UserPublicationsFilters, I.GetUserParameters>
 ) => {
     try {
-        // Authenticated account owners can retrieve private fields (such as email)
         const isAccountOwner = Boolean(event.user?.id === event.pathParameters.id);
 
-        const statuses: Array<I.ValidStatuses> = isAccountOwner ? ['DRAFT', 'LIVE'] : ['LIVE'];
-
-        const userPublications = await userService.getPublications(event.pathParameters.id, statuses, isAccountOwner);
+        const user = isAccountOwner ? event.user : await userService.get(event.pathParameters.id);
 
-        if (!userPublications) {
-            return response.json(404, { message: 'User not found' });
+        if (!user) {
+            return response.json(400, { message: 'Invalid user id' });
```

Comment:
I think 404 was correct  for this case, 400 would be correct if the 'user' part of the path had to match a particular pattern , but we currently allow anything here

Reply:
404 is not correct here, because we're not querying the user itself, it's more of a client error: 400 bad request, because the client should know exactly the id of the user it's trying to get the publications for, it's not like guessing.

Reply:
that is not what 400 is for, you should be able to determine if you need to return a 400 by purely examining the request and not by e.g calling a service to look up a user

Reply:
If a user is deleted their id is still 'valid' - if someone has bookmarked the user's profile page that should then return 404  (or 410) not 400, and if someone types arandom user id  that does not make it a bad request (and the client has no way of knowing it is not a 'real' user)

Reply:
https://github.com/paralleldrive/cuid userids are cuids so we _could_ put validation in that the the userid has to start 'c' but that's about it

Reply:
we'll leave it as 400 but maybe change the message to 'User not found'...

Reply:
changed the message back to 'User not found'